src/expr/src/relation/func.rs

```diff
@@ -580,10 +580,39 @@ fn lag_lead_inner_respect_nulls<'a>(
     result
 }
 
+#[allow(clippy::as_conversions)]
```

Comment:
Can you add a comment justifying this `allow`?

Reply:
Will do! `i64` indexes get involved in this function because it's convenient to allow negative indexes and have `datums_get` fail on them, and thus handle the beginning and end of the input vector uniformly, rather than checking underflow separately during index manipulations.